packages/api/src/external/hie/__tests__/set-doc-query-progress.test.ts

```diff
@@ -334,4 +344,315 @@ describe("aggregateAndSetHIEProgresses", () => {
       requestId,
     });
   });
+
+  it("has download completed when overallDocQueryProgress has download processing, externalData is empty", async () => {
+    const externalData: PatientExternalData = {};
+
+    const overallDocQueryProgress: DocumentQueryProgress = {
+      ...docQueryProgress,
+      convert: undefined,
+    };
+
+    const aggregateAndSetHIEProgressesResult = aggregateAndSetHIEProgresses(
+      overallDocQueryProgress,
+      externalData
+    );
+
+    expect(aggregateAndSetHIEProgressesResult).toEqual({
+      download: { total: 0, errors: 0, status: "completed", successful: 0 },
+      requestId,
+    });
```

Comment:
It seems we're doing some things and passing some values so the test is successful, but things that doesn't seem related to what we're actually testing; e.g.:
- setting convert to undefined on line 353
- passing total, errorr, successful on line 362

We can avoid that by using `expect.objectContainig`, like so:

```ts
    expect(aggregateAndSetHIEProgressesResult).toEqual(
      expect.objectContaining({
        download: expect.objectContaining({ status: "completed" }),
        requestId,
      })
    );
```

This allows us to validate certain properties without worrying about the rest of the object's state. It makes testing easier to write and to understand.

Like, the current version, I'm wondering why we're validating total to be zero if the test seem related to the status alone.